cognite_toolkit/cdf_tk/load.py

```diff
@@ -313,120 +197,260 @@ def load_transformations(
+        return TimeSeriesAcl(
+            [TimeSeriesAcl.Action.Read, TimeSeriesAcl.Action.Write],
+            TimeSeriesAcl.Scope.DataSet([ToolGlobals.data_set_id]),
+        )
+
+    @classmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError("Datapoints do not have an id")
+
+    def create(
+        self, items: pd.DataFrame | Sequence[pd.DataFrame], ToolGlobals: CDFToolConfig, drop: bool
+    ) -> pd.DataFrame | pd.DataFrame | None:
+        for item in items:
+            self.client.time_series.data.insert_dataframe(item)
+        return None
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> T_Resource:
+        if filepath.suffix == ".csv":
+            return pd.read_csv(filepath, parse_dates=True, index_col=0)
+        elif filepath.suffix == ".parquet":
+            return pd.read_parquet(filepath, engine="pyarrow")
         else:
-            print(f"  Would have created {len(transformations)} transformation.")
-    except Exception as e:
-        print(f"[bold red]ERROR:[/] Failed to create transformations.\n{e}")
-        ToolGlobals.failed = True
+            raise ValueError(f"Not supported file type {filepath.suffix}")
+
+
+@final
+class RawLoader(Loader[str, pd.DataFrame, list[pd.DataFrame]]):
+    load_files_individually = True
+    filetypes = frozenset({"csv", "parquet"})
+    api_name = "raw.rows"
+    folder_name = "raw"
+    resource_cls = pd.DataFrame
+    actions = frozenset({})
+    acl = RawAcl
+    default_db: str = "default"
+
+    def __init__(self, client: CogniteClient):
+        super().__init__(client)
+        self.db = self.default_db
+        self.table = ""
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        return RawAcl([RawAcl.Action.Read, RawAcl.Action.Write], RawAcl.Scope.All())
+
+    @classmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError("Raw data does not have an id")
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> pd.DataFrame:
+        for pattern in [r"(\d+)\.(\w+)\.(\w+)", r"(\d+)\.(\w+)"]:
+            if match := re.match(pattern, filepath.name):
+                self.db = match.group(2) if len(match.groups()) == 3 else self.default_db
+                self.table = match.group(len(match.groups()))
+                break
+        else:
+            print(f"[bold red]ERROR:[/] Filename {filepath.name} does not match expected format.")
+            ToolGlobals.failed = True
+            return pd.DataFrame()
+        if filepath.suffix == ".csv":
+            # The replacement is used to ensure that we read exactly the same file on Windows and Linux
+            file_content = filepath.read_bytes().replace(b"\r\n", b"\n").decode("utf-8")
+            df = pd.read_csv(io.StringIO(file_content), dtype=str)
+            df.fillna("", inplace=True)
+            return df
+        elif filepath.suffix == ".parquet":
```

Comment:
The additional support for parquet (other things?) should be added to the docs.

Reply:
Where do we have the different formats documented?